dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/ContainerStructureFinderStrategyResolverTest.java

```diff
@@ -56,4 +72,81 @@ public void get_container_structure_from_file_name_() throws Exception {
+
+        final ContainerStructureFinderStrategyResolver containerStructureFinderStrategyResolver =
+                new ContainerStructureFinderStrategyResolver();
+
+        final Container container = APILocator.getContainerAPI()
+                .getWorkingContainerByFolderPath(pathRoot, currentDefaultHost, APILocator.systemUser(), false);
+        final Optional<ContainerStructureFinderStrategy> containerStructureFinderStrategy =
+                containerStructureFinderStrategyResolver.get(container);
+
+        Assert.assertTrue(containerStructureFinderStrategy.isPresent());
+
+        final  List<ContainerStructure> containerStructures = containerStructureFinderStrategy.get().apply(container);
+
+        Assert.assertNotNull(containerStructures);
+        Assert.assertFalse(containerStructures.isEmpty());
+
+        final List<ContentType> contentTypes = APILocator.getContentTypeAPI(APILocator.systemUser()).findAll();
+        for (final ContentType contentType : contentTypes) {
+
+            Assert.assertTrue("Content Type: " + contentType.variable() + " does not included on the container",
+                    containerStructures.stream().anyMatch(containerStructure -> containerStructure.getStructureId().equals(contentType.id())));
+        }
+    }
+
+    private FileAsset createFileAsset (final String fileName1,
+                                       final String fileExtension,
+                                       final String fileContent,
+                                       final Folder root1,
+                                       final ContentType fileAssetContentType) throws Exception {
+
+        final FileAsset fileAsset = new FileAsset();
+        final File tempFile1      = File.createTempFile(fileName1, fileExtension);
+        FileUtil.write(tempFile1, fileContent);
+        final String fileNameField1 = fileName1 + fileExtension;
+
+        fileAsset.setContentType(fileAssetContentType);
+        fileAsset.setFolder(root1.getInode());
+        fileAsset.setBinary(FileAssetAPI.BINARY_FIELD, tempFile1);
+        fileAsset.setStringProperty(FileAssetAPI.HOST_FOLDER_FIELD, root1.getInode());
+        fileAsset.setStringProperty(FileAssetAPI.TITLE_FIELD, fileNameField1);
+        fileAsset.setStringProperty(FileAssetAPI.FILE_NAME_FIELD, fileNameField1);
+        fileAsset.setIndexPolicy(IndexPolicy.FORCE);
+
+        // Create a piece of content for the default host
+        return APILocator.getFileAssetAPI().fromContentlet(
+                APILocator.getContentletAPI().checkin(fileAsset, APILocator.systemUser(), false));
+    }
+
+    private Folder createAppContainerTest(final Host currentDefaultHost, final String pathRoot) throws DotDataException, DotSecurityException {
+
+        final Folder rootFolder = APILocator.getFolderAPI().createFolders(pathRoot, currentDefaultHost, APILocator.systemUser(), false);
+        return rootFolder;
```

Comment:
Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'rootFolder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6347242)